### PR TITLE
Minor edit

### PR DESCRIPTION
**Pull request by DCS78**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
CMake minimum - 3.20 => 3.29;
target_compile_features - 17 => 20;
updated external file versions to latest versions that does not cause an error;
blowfish.py - 2002 => 2015 version

## Steps to test these changes
Delete the project cache before rebuild. No changes ingame.

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
What is the issue you are trying to solve?

File: CMakeLists.txt [hunk 1]
```diff
@@ -135,4 +139,4 @@ endif()
 
 include_directories(${CMAKE_BINARY_DIR}/generated) # Globally include the build/generated directory
 
-add_subdirectory(src)
+add_subdirectory(src)
```
Comment (zach2good) [inline CMakeLists.txt]: Your editor is probably removing newlines on save. We keep newlines at the end of files.

File: CMakeLists.txt [hunk 2]
```diff
@@ -87,7 +91,7 @@ endif()
 
 # Link this 'library' to set the c++ standard / compile-time options requested
 add_library(project_options INTERFACE)
-target_compile_features(project_options INTERFACE cxx_std_17)
+target_compile_features(project_options INTERFACE cxx_std_20)
```
Comment (zach2good) [inline CMakeLists.txt]: 👍

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,10 +1,14 @@
-cmake_minimum_required(VERSION 3.20)
+cmake_minimum_required(VERSION 3.29)
```
Comment (zach2good) [inline CMakeLists.txt]: What do we gain by bumping this? We should try to keep tooling versions as low as reasonably possible unless we have a reason to update.

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 project(server C CXX)
 
 # https://cmake.org/cmake/help/latest/policy/CMP0069.html
 cmake_policy(SET CMP0069 NEW)
 set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
 
+# https://cmake.org/cmake/help/latest/policy/CMP0161.html
```
Comment (zach2good) [inline CMakeLists.txt]:
```
CMP0161
Added in version 3.29.

The [CPACK_PRODUCTBUILD_DOMAINS](https://cmake.org/cmake/help/latest/cpack_gen/productbuild.html#variable:CPACK_PRODUCTBUILD_DOMAINS) variable defaults to true.
```

We don't use CPack, why do we care about this?

File: ext/CMakeLists.txt [hunk 5]
```diff
@@ -68,13 +68,15 @@ endif(TRACY_ENABLE)
...
+    GIT_TAG 10.2.1 #e69e5f977d458f2650bb346dadf2ad30c5320281
+    VERSION 10.2.1 #11.0.2 - causes errors
 ) # defines: fmt::fmt
 
 CPMAddPackage(
     NAME spdlog
     GITHUB_REPOSITORY gabime/spdlog
-    GIT_TAG v1.12.0
+    GIT_TAG v1.14.1 #27cb4c76708608465c413f6d0e6b8d99a4d84302
+    VERSION 1.14.1
```
Comment (zach2good) [inline ext/CMakeLists.txt]: From the docs:
```
The origin may be specified by a GIT_REPOSITORY, but other sources, such as direct URLs, are [also supported](https://cmake.org/cmake/help/v3.11/module/ExternalProject.html#external-project-definition). If GIT_TAG hasn't been explicitly specified it defaults to v(VERSION), a common convention for git projects. On the other hand, if VERSION hasn't been explicitly specified, CPM can automatically identify the version from the git tag in some common cases. GIT_TAG can also be set to a specific commit or a branch name such as master, however this isn't recommended, as such packages will only be updated when the cache is cleared.
```

Is it redundant to specify both GIT_TAG and VERSION? I'd rather have just the tag and the comment with the commit hash, if we're not using just a specific hash

**Review by zach2good — COMMENTED** [review]
👋 Hello and welcome!

> tools/headlessxi/blowfish.py - moved to 2002 to 2015 version

If it works, why is this being changed?

**Comment by DCS78** [comment]
Thank you for the feedback.

This was the first commit I have ever made so I wanted to start small and test the waters.

There are, generally, two camps:
-"it works, why change it"
-"modernize"

I am attempting to unify the code base on .net 8 and remove some obsolete pieces. I realize my scope is narrow (only testing on Win11) so, it may not work globally I can put it out there for veteran programmers to decide.

As for why I bump versions. Usually, newer versions include optimizations as well as new features that maybe employed.

Partially, I am doing this to try to solve my issue of the map program randomly crashing. I have not pinned down a pattern, yet.

It is clear that I need to learn and adjust to community standards such as space at the end of files. I will look at my settings but I'm sure that was just me. As for the version, I will need to look again at the references that had it versus the ones that didn't to see what is different.

Again, thank you for the feedback.

Sent via the Samsung Galaxy S24 Ultra, an AT&T 5G smartphone


-------- Original message --------
From: Zach Toogood ***@***.***>
Date: 11/3/24 12:13 PM (GMT-08:00)
To: LandSandBoat/server ***@***.***>
Cc: Dave ***@***.***>, Author ***@***.***>
Subject: Re: [LandSandBoat/server] Minor edit (PR #6402)


@zach2good commented on this pull request.

👋 Hello and welcome!

tools/headlessxi/blowfish.py - moved to 2002 to 2015 version

If it works, why is this being changed?

________________________________

In CMakeLists.txt<https://github.com/LandSandBoat/server/pull/6402#discussion_r1827056161>:

> @@ -135,4 +139,4 @@ endif()

 include_directories(${CMAKE_BINARY_DIR}/generated) # Globally include the build/generated directory

-add_subdirectory(src)
+add_subdirectory(src)


Your editor is probably removing newlines on save. We keep newlines at the end of files.

________________________________

In CMakeLists.txt<https://github.com/LandSandBoat/server/pull/6402#discussion_r1827056173>:

> @@ -87,7 +91,7 @@ endif()

 # Link this 'library' to set the c++ standard / compile-time options requested
 add_library(project_options INTERFACE)
-target_compile_features(project_options INTERFACE cxx_std_17)
+target_compile_features(project_options INTERFACE cxx_std_20)


👍

________________________________

In CMakeLists.txt<https://github.com/LandSandBoat/server/pull/6402#discussion_r1827056444>:

> @@ -1,10 +1,14 @@
-cmake_minimum_required(VERSION 3.20)
+cmake_minimum_required(VERSION 3.29)


What do we gain by bumping this? We should try to keep tooling versions as low as reasonably possible unless we have a reason to update.

________________________________

In CMakeLists.txt<https://github.com/LandSandBoat/server/pull/6402#discussion_r1827056497>:

>  project(server C CXX)

 # https://cmake.org/cmake/help/latest/policy/CMP0069.html
 cmake_policy(SET CMP0069 NEW)
 set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)

+# https://cmake.org/cmake/help/latest/policy/CMP0161.html


CMP0161
Added in version 3.29.

The [CPACK_PRODUCTBUILD_DOMAINS](https://cmake.org/cmake/help/latest/cpack_gen/productbuild.html#variable:CPACK_PRODUCTBUILD_DOMAINS) variable defaults to true.


We don't use CPack, why do we care about this?

________________________________

In ext/CMakeLists.txt<https://github.com/LandSandBoat/server/pull/6402#discussion_r1827056783>:

> +    GIT_TAG v1.14.1 #27cb4c76708608465c413f6d0e6b8d99a4d84302
+    VERSION 1.14.1


From the docs:

The origin may be specified by a GIT_REPOSITORY, but other sources, such as direct URLs, are [also supported](https://cmake.org/cmake/help/v3.11/module/ExternalProject.html#external-project-definition). If GIT_TAG hasn't been explicitly specified it defaults to v(VERSION), a common convention for git projects. On the other hand, if VERSION hasn't been explicitly specified, CPM can automatically identify the version from the git tag in some common cases. GIT_TAG can also be set to a specific commit or a branch name such as master, however this isn't recommended, as such packages will only be updated when the cache is cleared.


Is it redundant to specify both GIT_TAG and VERSION? I'd rather have just the tag and the comment with the commit hash, if we're not using just a specific hash

**Pull request closed without merging**